packages/debug/src/browser/model/debug-thread.tsx

```diff
@@ -111,6 +113,26 @@ export class DebugThread extends DebugThreadData implements TreeElement {
         return this.session.sendRequest('pause', this.toArgs());
     }
 
+    get supportsGoto(): boolean {
+        return !!this.session.capabilities.supportsGotoTargetsRequest;
+    }
+
+    async jumpToCursor(uri: URI, position: monaco.Position): Promise<DebugProtocol.GotoResponse | undefined> {
+        const source = await this.session?.toDebugSource(uri);
+
+        if (!source) {
+            return undefined;
+        }
+
+        const response: DebugProtocol.GotoTargetsResponse = await this.session.sendRequest('gotoTargets', { source, line: position.lineNumber, column: position.column });
+
+        if (response && response.body.targets.length === 0) {
+            throw new Error('Unable to perform Jump to Cursor as there were no targets');
+        }
+        const targetId = response.body.targets[0].id;
```

Comment:
Suggestion: It would be great if you could implement a "choose the location" quick pick like it's done in [vscode](https://github.com/microsoft/vscode/blob/869a6422160a8716f09ae54f3ba6a767dc30bb05/src/vs/workbench/contrib/debug/browser/debugCommands.ts#L369) in case there are multiple targets.

Reply:
Hi @msujew , many thanks for your review, I could not find a debugger which supported multiple targets, as I could not test this feature, I decided not to implement the QuickInputService.pick here.